src/main/java/http/server/RequestRouter.java

```diff
@@ -25,18 +28,30 @@ public void routeRequest() {
             Request request = new Request(in);
             Response response = new Response(out);
             Handler handler = retrieveHandler(request);
-
-            handler.handle(response);
+            handler.setResponseValues(response);
             response.send();
         } catch (IOException err) {
             System.out.println(err.getMessage());
         }
     }
 
     public Handler retrieveHandler(Request request) {
-        Map<String, Handler> methodHandlers = handlers.get(request.getMethod());
-        return methodHandlers.containsKey(request.getPath()) ?
-                methodHandlers.get(request.getPath()) :
-                handlers.get("ERR").get("/not_found");
+        Map<String, Handler> pathHandlers = handlers.get(request.getPath());
+
+        if (pathExists(pathHandlers)) {
+            return methodExistsForPath(pathHandlers, request.getMethod()) ?
+                    pathHandlers.get(request.getMethod()) :
+                    new MethodNotAllowedHandler(pathHandlers);
```

Comment:
I'm not sure about a design where `MethodNotAllowedHandler` needs to accept `pathHandlers`.  I can see it is indeed used in `MethodNotAllowedHandler` , but are the allowed methods only specific to this handler? Can this logic be extracted from `MethodNotAllowedHandler`?